src/ReinforcementLearningZoo/src/algorithms/policy_gradient/util.jl

```diff
@@ -0,0 +1,63 @@
+using Distributions: DiscreteDistribution, ContinuousDistribution
+using Flux: softmax
+
+export action_distribution
+
+"""
+    action_distribution(dist, model_output)
+
+Compute the action distribution using the distribution type and output from a model.
+"""
+action_distribution(dist, model_output) =
+    throw(ArgumentError("dist ($dist) is not a ContinuousDistribution or DiscreteDistribution, not implemented"))
+
+"""
+    action_distribution(dist::Type{T}, model_output) where {T<:DiscreteDistribution}
+
+When `dist` is a subtype of `DiscreteDistribution`, assume `model_output` is a batch of unnormalized log probabilities.
+# Examples
+```jldoctest
+julia> model_output = reshape(1:10, 5, 2)
+5×2 reshape(::UnitRange{Int64}, 5, 2) with eltype Int64:
+ 1   6
+ 2   7
+ 3   8
+ 4   9
+ 5  10
+julia> action_distribution(Categorical, model_output)
+2-element Vector{Categorical{Float64, Vector{Float64}}}:
+ Categorical{Float64, Vector{Float64}}(
+support: Base.OneTo(5)
+p: [0.011656230956039605, 0.03168492079612427, 0.0861285444362687, 0.23412165725273662, 0.6364086465588308]
+)
+
+ Categorical{Float64, Vector{Float64}}(
+support: Base.OneTo(5)
+p: [0.011656230956039605, 0.03168492079612427, 0.0861285444362687, 0.23412165725273662, 0.6364086465588308]
+)
+```
+"""
+action_distribution(dist::Type{T}, model_output) where {T<:DiscreteDistribution} = 
+    map(col -> dist(col; check_args=false), eachcol(softmax(model_output)))
```

Comment:
bug in Zygote: if this were changed to `dist(col…)` we get zero gradients everywhere.

also note JuliaStats/Distributions.jl#1084